app/controllers/api/v1/targets_controller.rb

```diff
@@ -11,7 +11,19 @@ def index
       def create
         @target = current_user.targets.build(target_params)
         if @target.save
-          @target
+          match = search_match(@target.radius, @target.latitude, @target.longitude, @target.topic_id)
+          matched_user = {}
+          match_conversation = {}
+          
+          if match.size > 0
```

Comment:
I think you should iterate over match collection. The search could find multiple matches among users for the same topic.

Reply:
It is possible, but the requirement says one match. Only one.

Reply:
Yes, only one conversation between **_two_** users, but you can have a conversation with each user in the system. This function find every potencial user to create a match (conversation) with.

Reply:
Understood and solved!